### ISSUE TITLE
Vagrant box does not mount the host FrameworkBenchmark directory

### ISSUE THREAD

**Issue by cmsimike** [TechEmpower/FrameworkBenchmarks#1949]
I was under the assumption that I can just fire vagrant up and have it test my host's current working directory of FrameworkBenchmark. Turns out that is not the case and in order for me to test out any changes I make, I have to commit and push (possible incorrect code) to a repo, then pull in the VM. 

It was said that this is the case because Vagant does not work in Windows, which is fine, but why are we making it harder for osx and linux devs because Windows and Vagrant won't play well together?

I suggest making it work for the OSes that work well with vagrant. We can re-evaluate Windows as well to see if the situation has solved itself (or default to something like samba for windows)

**Comment by Zloy** [comment]
+1

**Comment by gmile** [comment]
I've ran into this as well.

As far as I see, the shared folder (one between your host OS and vagrant guest OS) is this one: [`/FrameworkBenchmarks`](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/deployment/vagrant-common/core.rb#L118). If you make changes to FB files on your host OS, they indeed get reflected in `/FrameworkBenchmarks`.

However `README.md` as well as [this](http://frameworkbenchmarks.readthedocs.io/en/latest/Development/Installation-Guide/) suggest one should work from within `~/FrameworkBenchmarks` (a sub-directory to your home directory).

I thought these two folders `/FrameworkBenchmarks` and might somehow be kept in sync in the background by some sort of utility scripts, yet I was unable to find any evidence this is true (actually I doubt this is true).

I will go one step further. Since `/FrameworkBenchmarks` indeed properly syncs with the outer world, I ran `./toolset/run-tests.py --mode verify --test your_framework` from within it only to see it fail.

The main observation is this: running `./toolset/run-tests.py --mode verify --test phoenix`

1. running `./toolset/run-tests.py --mode verify --test your_framework` **works** in `~/FrameworkBenchmarks`,
2. running `./toolset/run-tests.py --mode verify --test your_framework` **does not work** in `/FrameworkBenchmarks` (I can provide more details on the error if needed).

So yeah, this is confusing. @msmith-techempower, does the above make sense?

**Comment by nbrady-techempower** [comment]
@gmile what is your host OS? If it's indeed windows, then you'll have some line ending issues between editing on your host and running the tests on vagrant. However, there are still known issues to be worked out with the vagrant setup.

**Comment by gmile** [comment]
@nbrady-techempower it's OS X (10.11.3)

**Comment by nbrady-techempower** [comment]
@gmile thanks. Like I said, there's still some known issues here. We will look into getting the symbolic links working again as soon as possible. For now, if developing in the vagrant environment is cumbersome, we recommend forking the repo and developing in your host OS. Push the changes to your repo and pull them into vagrant for testing only.

**Comment by Zloy** [comment]
What if we [tweak Vagrantfile](http://stackoverflow.com/questions/26811089/vagrant-how-to-have-host-platform-specific-provisioning-steps) to get vagrant box same way for Windows host OS, but syncing TFB folder to ~/FrameworkBenchmarks for Ubuntu host OS?

In such case Linux users can fork the repo, clone it, get vagrant box running and develop in host OS. That would super simplify contributing for Linux users, leaving untouched current developing process for Windows users.

**Comment by gmile** [comment]
@Zloy I'm not sure it will help.

Yesterday I tried syncing FrameworkBenchmarks from host to `~/FrameworkBenchmarks` path on client (had to remove original `~/FrameworkBenchmarks` that appears during the initial box setup). Even though it synced, my framework tests didn't run.

Specifically, during the setup stage of the test I noticed some problems with permissions on synced folders, e.g. I couldn't set the test for my target framework (elixir/phoenix) due to the inability to create files (like installing erlang). Even simple `sed` failed for me (it attempts to create a file during it's "replacement operation").

> The bonus would be in eliminating cloning TFB to ~/FrameworkBenchmarks which takes quite some time
That'd be really nice to have, indeed.
